enonic/src/main/resources/site/parts/article-form/article-form.ts

```diff
@@ -47,9 +52,31 @@ function createThymeleafParams(res: QueryResponse<Comment>,article: Content<Arti
   return {
     title: article.data.title,
     body: article.data.body,
-    comments: res.hits.map((content: Content<Comment>) => content.data),
+    comments: createCommentTree(res.hits),
     serviceUrl: serviceUrl('comment'),
     parentPath: article._path,
     pathSelf: path
   }
+}
+
+const byParentFirst = pipe(ordBoolean, contramap((c: Content<Comment>) => !!c.data.parentId))
+
+export function childrenToParent (acc: ContentTree<Comment>[], curr: Content<Comment>): ContentTree<Comment>[] {
+  if(curr.data.parentId) {
+    const parent = acc.filter((item) => item._id === curr.data.parentId);
```

Comment:
`parent` is here an array, so the name should either be pluralized, or we should pick out the entry.
```suggestion
    const parent = acc.filter((item) => item._id === curr.data.parentId)[0];
```

Reply:
Changed